osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs

```diff
@@ -0,0 +1,399 @@
+                            {
+                                case PathType.Bezier:
+                                    writer.Write("B|");
+                                    break;
+
+                                case PathType.Catmull:
+                                    writer.Write("C|");
+                                    break;
+
+                                case PathType.PerfectCurve:
+                                    writer.Write("P|");
+                                    break;
+
+                                case PathType.Linear:
+                                    writer.Write("L|");
+                                    break;
+                            }
+
+                            lastType = point.Type.Value;
+                        }
+                        else
+                        {
+                            // New segment with the same type - duplicate the control point
+                            writer.Write(FormattableString.Invariant($"{positionData.X + point.Position.Value.X}:{positionData.Y + point.Position.Value.Y}|"));
+                        }
+                    }
+
+                    if (i != 0)
+                    {
+                        writer.Write(FormattableString.Invariant($"{positionData.X + point.Position.Value.X}:{positionData.Y + point.Position.Value.Y}"));
+                        writer.Write(i != curveData.Path.ControlPoints.Count - 1 ? "|" : ",");
+                    }
+                }
+
+                writer.Write(FormattableString.Invariant($"{curveData.RepeatCount + 1},"));
+                writer.Write(FormattableString.Invariant($"{curveData.Path.Distance},"));
+
+                for (int i = 0; i < curveData.NodeSamples.Count; i++)
+                {
+                    writer.Write(FormattableString.Invariant($"{(int)toLegacyHitSoundType(curveData.NodeSamples[i])}"));
+                    writer.Write(i != curveData.NodeSamples.Count - 1 ? "|" : ",");
+                }
+
+                for (int i = 0; i < curveData.NodeSamples.Count; i++)
+                {
+                    writer.Write(getSampleBank(curveData.NodeSamples[i], true));
+                    writer.Write(i != curveData.NodeSamples.Count - 1 ? "|" : ",");
+                }
+            }
+            else if (hitObject is IHasEndTime endTimeData)
+                writer.Write(FormattableString.Invariant($"{endTimeData.EndTime},"));
+
+            writer.Write(hitObject is IHasCurve
+                ? getSampleBank(hitObject.Samples, zeroBanks: true)
+                : getSampleBank(hitObject.Samples));
+
+            writer.Write(Environment.NewLine);
+        }
+
+        private void handleTaikoHitObject(TextWriter writer, HitObject hitObject)
+        {
+        }
+
+        private void handleCatchHitObject(TextWriter writer, HitObject hitObject)
+        {
+        }
+
+        private void handleManiaHitObject(TextWriter writer, HitObject hitObject)
+        {
+        }
```

Comment:
Could at least throw `NotImplemented` for now maybe? Unless you're aiming for fast integration with standard editor in which case I guess not throwing is slightly more preferable.